prebid/native-implementation.md

```diff
@@ -0,0 +1,441 @@
+| adTemplate | optional | Used in the ‘AdUnit-Defined Creative Scenario’, this value controls the Native template right in the page. | See [example](#5-implementing-adunit-defined-creative) below. | escaped ES5 string |
+| rendererUrl | optional | Used in the ‘Custom Renderer Scenario’, this points to javascript code that will produce the Native template. | 'https://host/path.js' | string |
+| type | optional | A “type” is like a macro that defines a group of assets. The only value currently supported is ‘image’, which implies the following assets: image, title, sponsoredBy, clickUrl, body, icon, and cta. The first 4 are required attributes. | `image` | string |
+| ASSETCODE. required | optional | Defines whether native bids must include this asset. Defaults to `false`. | `true` | boolean |
+| ASSETCODE. len | optional | For text assets, bidders should pass this value through to the endpoint. Prebid.js does not enforce the responses and there’s no default. | 40 | integer |
+| ASSETCODE. sizes | optional | For image assets, bidders may pass this value through to the endpoint. Prebid.js does not enforce the responses and there’s no default. Format is an array with two elements: [WIDTH, HEIGHT] | `[50, 50]` | array of integers |
+| ASSETCODE. aspect_ratios | optional | For image assets, bidders may pass this value through to the endpoint. Prebid.js does not enforce the responses and there’s no default. Format is an object with the attributes in the rows below.  | | object |
+| ASSETCODE. aspect_ratios.min_width | optional | Part of the aspect_ratios object, bidders may pass this value through to the endpoint. Prebid.js does not enforce the responses and there’s no default. | 50 | integer |
+| ASSETCODE. aspect_ratios.min_height | optional | Part of the aspect_ratios object, bidders may pass this value through to the endpoint. Prebid.js does not enforce the responses and there’s no default. | 75 | integer |
+| ASSETCODE. aspect_ratios.ratio_width | optional | Part of the aspect_ratios object, bidders may pass this value through to the endpoint. Prebid.js does not enforce the responses and there’s no default. | 2 | integer |
+| ASSETCODE. aspect_ratios.ratio_height | optional | Part of the aspect_ratios object, bidders may pass this value through to the endpoint. Prebid.js does not enforce the responses and there’s no default. | 3 | integer |
+
+
+**Table 3: Native Assets Recognized by Prebid.js**
+
+{% include dev-docs/native-assets.md %}
+
+{: .alert.alert-warning :}
+Specific bidders may not support all of the fields listed below or may return differing responses for the assets that are requested.
+
+### Two ways to define image sizes
+
+{% include dev-docs/native-image-asset-sizes.md %}
+
+## 4. Implementing AdServer-Defined Creative
```

Comment:
I feel that the index can be confusing as we expect the Publisher to choose one of the three:

## 4. Implementing AdServer-Defined Creative
## 5. Implementing AdUnit-Defined Creative
## 6. Implementing the Custom Renderer Scenario

Maybe:

## 4.1 Implementing AdServer-Defined Creative
## 4.2 Implementing AdUnit-Defined Creative
## 4.3 Implementing the Custom Renderer Scenario